src/rgw/rgw_web_idp.h

```diff
@@ -0,0 +1,51 @@
+#ifndef CEPH_RGW_WEB_IDP_H
+#define CEPH_RGW_WEB_IDP_H
+
+#include <utility>
+#include <boost/optional.hpp>
+#include <boost/utility/string_view.hpp>
+
+#include "rgw_auth.h"
+#include "rgw_common.h"
+
+namespace rgw {
+namespace web_idp {
+
+//WebToken contains some claims from the decoded token which are of interest to us.
+class WebTokenClaims {
```

Comment:
this could probably just be a struct with public member variables, if you're exposing them all with getters/setters anyway